yt/utilities/lib/particle_mesh_operations.pyx

```diff
@@ -83,88 +83,97 @@ def CICDeposit_3(np.ndarray[np.float64_t, ndim=1] posx,
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def CICDeposit_2(np.ndarray[np.float64_t, ndim=1] posx,
-                 np.ndarray[np.float64_t, ndim=1] posy,
-                 np.ndarray[np.float64_t, ndim=1] mass,
+@cython.cdivision(True)
+def CICDeposit_2(np.float64_t[:] posx,
+                 np.float64_t[:] posy,
+                 np.float64_t[:] mass,
                  np.int64_t npositions,
-                 np.ndarray[np.float64_t, ndim=2] field,
-                 np.ndarray[np.float64_t, ndim=1] leftEdge,
-                 np.ndarray[np.int32_t, ndim=1] gridDimension,
-                 np.ndarray[np.float64_t, ndim=1] cellSize):
+                 np.float64_t[:, :] field,
+                 np.float64_t[:] x_bin_edges,
+                 np.float64_t[:] y_bin_edges):
 
     cdef int i1, j1, n
     cdef np.float64_t xpos, ypos
-    cdef np.float64_t edge0, edge1
-    cdef np.float64_t le0, le1
-    cdef np.float64_t dx, dy, dx2, dy2
+    cdef np.float64_t edgex, edgey
+    cdef np.float64_t dx, dy, ddx, ddy, ddx2, ddy2
 
-    edge0 = (<np.float64_t> gridDimension[0]) - 0.5001
-    edge1 = (<np.float64_t> gridDimension[1]) - 0.5001
+    edgex = (<np.float64_t> x_bin_edges.shape[0]) + 0.5001
+    edgey = (<np.float64_t> y_bin_edges.shape[0]) + 0.5001
 
-    le0 = leftEdge[0]
-    le1 = leftEdge[1]
+    # We are always dealing with uniformly spaced bins for CiC
+    dx = x_bin_edges[1] - x_bin_edges[0]
+    dy = y_bin_edges[1] - y_bin_edges[0]
 
     for n in range(npositions):
 
         # Compute the position of the central cell
-        xpos = (posx[n] - le0)/cellSize[0]
-        ypos = (posy[n] - le1)/cellSize[1]
+        xpos = (posx[n] - x_bin_edges[0])/dx
+        ypos = (posy[n] - y_bin_edges[0])/dy
 
-        if (xpos < 0.5001) or (xpos > edge0):
+        if (xpos < -0.5001) or (xpos > edgex):
```

Comment:
Can you explain this change?  Is this because of the change above to definitions of `edgex` and `edgey` now being `+ 0.5001` ?  Where did that come from?